ARCs/arc-0028.md

```diff
@@ -0,0 +1,81 @@
+---
+arc: 28
+title: Algorand Event Log Spec
+status: Draft
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/{TBD}
+---
+
+# Algorand Event Log Spec
+
+> This API is a draft.
+> Some elements may change.
+
+## Summary
+
+A methodology for structured logging by Algorand dapps.
+
+## Abstract
+
+Algorand dapps can use the `log` primitive to attach information about an application call. This ARC proposes the concept of Events, which are merely a way in which data contained in these logs may be categorized and structured.
+
+In short: to emit an Event, a dapp calls `log` with ABI formatting of the log data, and a 4-byte prefix to indicate which Event it is.
+
+## Specification
+
+Each kind of Event emitted by a given dapp has a unique 4-byte identifier. This identifier is derived from its name and the structure of its contents, like so:
+
+### Event Signature
+
+An Event Signature is a utf8 string, comprised of: the name of the event, followed by an open paren, followed by the comma-separated names of the data types contained in the event, followed by a close paren. This follows naming conventions similar to ABI signatures, but does not include the return type.
```

Comment:
Do you want to explicitly say that the data types allowed are the same as ARC-4? Perhaps explicitly _disallowing_ the reference types like `asset` and `txn`?